[Enhancement] Add header setting support to http_client

Nboxff opened the pull request:
## Why I'm doing:

Currently, the HTTPClient class lacks the ability to set custom headers, which limits its functionality for various API interactions such as calling large language models' service.

## What I'm doing:

add methods in http_client.h and http_client.cpp

```
// Set HTTP header
void set_header(const std::string& key, const std::string& value);

// Set multiple headers at once
void set_headers(const std::unordered_map<std::string, std::string>& headers);

// Clear all headers
void clear_headers();

// Set Authorization header with Bearer token (useful for LLM APIs)
void set_bearer_token(const std::string& token) { set_header("Authorization", "Bearer " + token); }
```
## What type of PR is this:

- [ ] BugFix
- [ ] Feature
- [x] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

Does this PR entail a change in behavior?

- [ ] Yes, this PR will result in a change in behavior.
- [x] No, this PR will not result in a change in behavior.

If yes, please specify the type of change:

- [ ] Interface/UI changes: syntax, type conversion, expression evaluation, display information
- [ ] Parameter changes: default values, similar parameters but with different default values
- [ ] Policy changes: use new policy to replace old one, functionality automatically enabled
- [ ] Feature removed
- [ ] Miscellaneous: upgrade & downgrade compatibility, etc.

## Checklist:

- [x] I have added test cases for my bug fix or my new feature
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function
- [ ] This is a backport pr

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [ ] 3.5
  - [ ] 3.4
  - [ ] 3.3

Nboxff committed:
[Enhancement] Add header setting support to http_client for LLM integration

Signed-off-by: xiebofan <1814739992@qq.com>

Nboxff committed:
[Refactor] Fix EOL for http_client.h and http_client.cpp

Signed-off-by: xiebofan <1814739992@qq.com>

alvin-celerdata commented:
@cursor review

alvin-celerdata commented:
@Nboxff could you sign-off it by `git commit -s` to make the DCO check pass

Nboxff committed:
[Refactor] Remove unnecessary statements in http_client and enhance the unit test

- Use the insert_or_assign method instead of []
- Remove redundant _apply_headers method call
- Enhance unit test for http_client

Signed-off-by: xiebofan <1814739992@qq.com>

Nboxff commented on be/src/http/http_client.cpp:
```diff
@@ -138,6 +138,44 @@ void HttpClient::set_method(HttpMethod method) {
+        curl_slist_free_all(_header_list);
+        _header_list = nullptr;
+    }
+    _apply_headers();
```
Thanks for the thorough review! `_apply_headers()` is unnecessary and I have removed it in the latest commit.

Nboxff commented on be/test/http/http_client_test.cpp:
```diff
@@ -69,8 +69,42 @@ class HttpClientTestSimplePostHandler : public HttpHandler {
+        }
+
+        // Check authentication header type
+        const auto& auth_header = req->header(HttpHeaders::AUTHORIZATION);
```
Thanks! I have updated tests to echo Authorization header in response body and update the testcases.

Nboxff committed:
[UT] Remove the assertion that accesses the private member variable _headers

Signed-off-by: xiebofan <1814739992@qq.com>

Nboxff commented:
> @Nboxff could you sign-off it by `git commit -s` to make the DCO check pass你能通过 `git commit -s` 签字，让 DCO 检查通过吗

Thanks for the reminder! I've signed off all commits and the DCO check should now pass.

alvin-celerdata commented on be/src/http/http_client.cpp:
```diff
@@ -168,6 +205,10 @@ Status HttpClient::execute_delete_request(const std::string& payload, std::strin
     _callback = &callback;
+
+    // Apply headers before executing
+    _apply_headers();
```
_apply_headers is called when set_headers is executed every time. So there is no need to call _apply_headers in this place.

And init() will be called when call execute, so _headers should be reset in init function to be consistent with _header_list.

Nboxff committed:
[Refactor] Reset _headers in HttpClient::init() and remove unnecessary _apply_headers call

- Reset _headers in HttpClient::init()
- add curl_easy_setopt in clear_headers
- remove unnecessary _apply_headers call in execute()

Signed-off-by: xiebofan <1814739992@qq.com>
